src/NuGet.Core/NuGet.Protocol/Converters/PackageDependencyGroupConverter.cs

```diff
@@ -17,11 +15,31 @@ public class PackageDependencyGroupConverter : JsonConverter
 
         public override bool CanWrite => false;
 
+        private readonly PackageDependencyConverter _converter = new PackageDependencyConverter();
+
         public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
-            var set = JsonUtility.LoadJson(reader);
```

Comment:
The current implementation of [`JsonUtility.LoadJson`](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol/Utility/JsonUtility.cs#L35-L38) has the following settings when loading the JSON file into a `JObject`. Can we confirm that the changes proposed in the PR have the same functionality (e.g., comments are ignored, etc.)?

https://github.com/NuGet/NuGet.Client/blob/7ca037b450fffbae993f65367e29b9c37da7a012/src/NuGet.Core/NuGet.Protocol/Utility/JsonUtility.cs#L15-L19

Reply:
Those settings don't exist on the [JsonSerializerSettings](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_JsonSerializerSettings.htm) class. Seem strange to mix these two implementations. Is PackageDependencyGroup known to have comments?

Reply:
no, it's not known to have comments, but if it does, it's out of control of the customer if NuGet fails because of it.

Reply:
At the end I'm using the settings we have in https://github.com/NuGet/NuGet.Client/blob/7856ac6d46ac73374ef24e6fa637283cd83ae54e/src/NuGet.Core/NuGet.Protocol/Converters/JsonExtensions.cs#L19-L35

Reply:
If we can add a new unit or modify existing ones to ensure that they pass even though the JSON content has a comment then I guess we should be good.

Reply:
JSON objects cannot have a comments https://github.com/JamesNK/Newtonsoft.Json/issues/1315#issuecomment-302036211

Reply:
That's for JObject. But James mentioned that in the same comment that `If you need to access the comment value then you will need to use the lower level JsonTextReader class to read it.`  If my understanding is correct, we are using lower-level reader here to deserialize JSON.

Reply:
Creating a console app and adding Newtonsoft.Json to it, I use this test data:

```cs
string json = @"{
""key"":""myKey"",
//  comment
""value"": ""myValue""
}";
```

Using `var jToken = JToken.Parse(json);` doesn't throw.

Using `var kvp = Newtonsoft.Json.JsonConvert.DeserializeObject<KeyValuePair<string, string>>(json);` does throw, and I couldn't figure out what `JsonSerializerSettings` to pass in to make it work, within 30 seconds.

However, this does not throw with System.Text.Json:
```cs
System.Text.Json.JsonSerializerOptions options = new()
{
    ReadCommentHandling = System.Text.Json.JsonCommentHandling.Skip
};
System.Text.Json.JsonSerializer.Deserialize<KeyValuePair<string, string>>(json, options);
```

So, if we can't figure out how to make Newtonsoft.Json's `JsonSerializer` ignore the comments, maybe we should put in the extra effort to port this to System.Test.Json instead. We're eventually going to have to anyway.

Reply:
Sorry, I got busy with other things.

I'll try to change this to use `System.Test.Json`, but I think that is not super important, this is being used when we are deserializing the json from the registration api, I don't believe we will ever encounter a comment there

Reply:
@martinrrm I have unresolved this conversation because you mentioned that `I don't believe we will ever encounter a comment in the registration API`. We were ignoring comments in the past. I am concerned that if this change might cause any regression if there are comments in the JSON payload.

As suggested, if we can add a new unit test or modify existing ones to ensure that they pass even though the JSON content has a comment, then I guess we should be good.

Reply:
I see that you have already modified tests by adding comments in the JSON payload https://github.com/NuGet/NuGet.Client/pull/5872/files#diff-5d4614874e872a04258bc12705f69b82e95e76927320b5b0e8e2daa3dac1ccdcR303. I think we are good.